jicoco-kotlin/src/main/kotlin/org/jitsi/config/LegacyConfigFileLoader.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.config
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import org.jitsi.utils.logging2.LoggerImpl
+import java.nio.file.InvalidPathException
+import java.nio.file.Paths
+
+/**
+ * Used to load a config file that exists outside of the typesafe config
+ * library's convention for log file locations.  For example, we use this to
+ * load a legacy sip-communicator.properties config file into a [Config]
+ * instance.
+ */
+class LegacyConfigFileLoader {
+    companion object {
+        private val logger = LoggerImpl(LegacyConfigFileLoader::class.java.name)
+
+        fun load(first: String?, vararg rest: String?): Config {
```

Comment:
Can we add docs for `first` and `rest`?

Reply:
added a doc here.  while in there i started wondering if i didn't need to split them there (and just take a single `varargs pathParts: String?`, but it seemed like it's just 6-one-way half-dozen-the-other :) so left it as-is.